### DISCUSSION TITLE
ESP32 & SH1106 Troubleshooting

### DISCUSSION THREAD

**Discussion by rgstephens** [arendst/Tasmota#11937] [Q&A] [answered]
I have an ESP32 with an SH1106 from [aliexpress](https://www.aliexpress.com/item/32807898232.html?spm=a2g0s.9042311.0.0.27424c4dHJ4WpW) and I can't get the display to work. I'm running 9.4.0 tasmota32-display. Are there some general troubleshooting steps I should follow?

`i2cscan` shows no devices.

`display` shows:

```json
{"Display":{"Model":2,"Type":0,"Width":128,"Height":64,"Mode":3,"Dimmer":53,"Size":1,"Font":1,"Rotate":0,"Invert":0,"Refresh":2,"Cols":[16,8],"Rows":2}}
```

I've tried several `displaymode` options and `displaytext abc` but see nothing.

**Comment by sfromis** [comment, marked as answer]
Since I2CScan shows no devices (I'm assuming `{"I2CScan":"No devices found"}`), you need to make sure to config I2C on the correct pins. Until it shows up on I2C, Tasmota can do nothing.

**Reply by rgstephens** [reply]
Thanks, set `GPIO4 = I2C SCL` & `GPIO5 = I2C SDA`, now `i2cscan` now shows:

```json
{"I2CScan":"Device(s) found at 0x3c"}
```

Still don't see any text after trying various `displaymode` and `displaytext` commands.

**Reply by sfromis** [reply]
You may have to power on the display also. At least with the somewhat similar SSD1306, Tasmota gets a power toggle on the front page (and a command of course).

**Reply by rgstephens** [reply]
yes, there's the front page power toggle and command. Didn't realize this might be related to display power but it still isn't displaying anything. Seems like `i2cscan` wouldn't have seen the device at `0x3c` if it wasn't getting power.

**Reply by sfromis** [reply]
No, power is power to the display, not to the controller chip. Tasmota can command the controller chip to turn on and off the display. I2CScan should always detect the display, with the pixels on or off.

**Reply by sfromis** [reply]
Another thing to check is DisplayAddress. In some instances, I've seen a wrong address being used. The correct one is decimal 60, matching the I2C address of 0x3c. If I hook up a SSD1306 with a binary compiled to include `USE_DISPLAY_SSD1306` (like the precompiled tasmota-displays), set `DisplayMode 1` and make sure that it is powered on, I'm getting date and time on the screen. You may have to fiddle with the various parameters visible in the `Display` command output to adjust font, size, rotation and lines/cols.

**Reply by rgstephens** [reply]
That was it, thanks!  DisplayAddress showed 113, changing it to 60 fixed it.
